### PR TITLE
ALF-22007 fix TransactionListeners execution order

### PR DESCRIPTION
**Pull request by nbarithel**
fix and unit test.

### CODE REVIEW

File: src/test/java/org/alfresco/util/transaction/SpringAwareUserTransactionTest.java [hunk 1]
```diff
@@ -321,6 +322,84 @@ private UserTransaction getFailingTxn()
...
+            catch (Exception ee)
+            {
+                e.addSuppressed(ee);
+            }
+            throw e;
+        }
+        checkNoStatusOnThread();
+    }
+    
+    private static class TestTransactionListerner extends TransactionListenerAdapter
```
Comment (derekhulley) [inline src/test/java/org/alfresco/util/transaction/SpringAwareUserTransactionTest.java]: Minor typo

File: src/test/java/org/alfresco/util/transaction/SpringAwareUserTransactionTest.java [hunk 2]
```diff
@@ -321,6 +322,84 @@ private UserTransaction getFailingTxn()
...
+    {
+        private final String name;
+        private final StringBuffer buffer;
+        
+        public TestTransactionListerner(String name, StringBuffer buffer) {
```
Comment (derekhulley) [inline src/test/java/org/alfresco/util/transaction/SpringAwareUserTransactionTest.java]: Minor formatting.

**Comment by derekhulley** [comment]
Thanks.  This makes sense.

**Comment by nbarithel** [comment]
Great ! Fixed typo and formatting.

**Comment by mmuller88** [comment]
Thank you very much for providing this awesome fix. Accordingly to our internally workflow I will create a new test branch for testing your fix and if everything goes fine I will merge it.
Cheers Martin

**Pull request merged**